src/UsgsAstroSarSensorModel.cpp

```diff
@@ -136,3 +140,149 @@ string UsgsAstroSarSensorModel::constructStateFromIsd(
   // some state data is not in the ISD and requires a SM to compute them.
   return state.dump();
 }
+
+// TEMPORARY hard-coded values. Need to be replaced by accessors.
+double START_TIME = 0.0;
+double STOP_TIME = 5.0;                         
+double EXPOSURE_DURATION = 0.005;
+double SCALED_PIXEL_WIDTH = 7.5; 
+double WAVELENGTH = 0.125;
+double SAMPLES = 1000.0;
+double LINES = 1000.0;
+
+csm::ImageCoord UsgsAstroSarSensorModel::groundToImage(
+    const csm::EcefCoord& groundPt,
+    double desiredPrecision,
+    double* achievedPrecision,
+    csm::WarningList* warnings) const {
+
+  //MESSAGE_LOG(this->m_logger, "Computing groundToImage(ImageCoord) for {}, {}, {}, with desired precision {}",
+  //          groundPt.x, groundPt.y, groundPt.z, desiredPrecision);
+
+  // Find time of closest approach to groundPt and the corresponding slant range by finding 
+  // the root of the doppler shift frequency
+  csm::EcefCoord newGroundPt(groundPt);
+  double time = dopplerShiftRoot(newGroundPt);
+  double slantRange = calcSlantRange(newGroundPt, time);
+
+  // Find the ground range, based on the ground-range-to-slant-range polynomial defined by the 
+  // range coefficient set, with a time closest to the calculated time of closest approach
+  double groundRange = slantToGroundRange(groundPt, time, slantRange);
+
+  double line = (time - START_TIME) / EXPOSURE_DURATION;
+  double sample = groundRange / SCALED_PIXEL_WIDTH;
+  return csm::ImageCoord(line, sample);
+
+  throw csm::Error(
+    csm::Error::UNKNOWN_ERROR,
+    "Could not calculate groundToImage.",
+    "groundToImage");
```

Comment:
Why is this here? This will never run because of the return call.

Reply:
Hey, thanks for catching this. I think this used to be in an 'else' but I've since moved the exceptions to the functions that that this function calls. I'm going to add a try/catch & rethrow this error in here.